GEOSwift/GeoJSON.swift

```diff
@@ -149,13 +149,26 @@ private func ParseGEOJSONFeature(_ GEOJSONFeature: Dictionary<String, AnyObject>
     // map feature representaion to Feature Object with properties and GEOS geometry
     if let geometry = GEOJSONFeature["geometry"] as? Dictionary<String,AnyObject>,
         let properties = GEOJSONFeature["properties"] as? NSDictionary,
-        let geometryType = geometry["type"] as? String,
-        let geometryCoordinates = geometry["coordinates"] as? NSArray,
-        let geometryObject = ParseGEOJSONGeometry(geometryType, coordinatesNSArray: geometryCoordinates) {
-            let feature = Feature()
-            feature.geometries?.append(geometryObject)
-            feature.properties = properties
-            return feature
+        let geometryType = geometry["type"] as? String {
+            if geometryType != "GeometryCollection" {
+                if let geometryCoordinates = geometry["coordinates"] as? NSArray,
+                let geometryObject = ParseGEOJSONGeometry(geometryType, coordinatesNSArray: geometryCoordinates) {
+                    let feature = Feature()
+                    feature.geometries?.append(geometryObject)
+                    feature.properties = properties
+                    feature.data = GEOJSONFeature
+                    return feature
+                }
+            } else {
+                if let geometryCollection = geometry["geometries"] as? NSArray,
+                    let geometryObject = ParseGEOJSONGeometryCollection(geometryCollection) {
+                    let feature = Feature()
+                    feature.geometries = geometryObject
+                    feature.properties = properties
+                    feature.data = GEOJSONFeature
+                    return feature
+                }
+            }
```

Comment:
We should factor our the feature creation and configuration. It's the same in both cases.